openmdao/solvers/nonlinear/newton.py

```diff
@@ -68,13 +68,17 @@ def _declare_options(self):
         """
         super(NewtonSolver, self)._declare_options()
 
-        self.options.declare('solve_subsystems', types=bool, default=False,
```

Comment:
I think there is behavior missing here. If the user does pass a value for "solve_subsystems" as an attribute into the constructor, then there should be a DeprecationWarning raised that tells the user that in openmdao 3.x and beyond, the "solve_subsystems" argument is going to be required.  Then, to conserve backwards compatibility, we should set it to False.

Also, we need a deprecation test that confirms this.